core/src/main/java/com/snowgears/shop/hook/WorldGuardHook.java

```diff
@@ -49,13 +51,25 @@ private static class Internal {
 
         public static void registerAllowShopFlag(Shop plugin) {
             Bukkit.getLogger().log(Level.INFO,"[Shop] Registering WorldGuard flag '" + FLAG_ALLOW_SHOP + "'");
+            FlagRegistry registry = WorldGuard.getInstance().getFlagRegistry();
+            // see https://worldguard.enginehub.org/en/latest/developer/regions/custom-flags/
             try {
-                StateFlag allowShopFlag = new StateFlag(FLAG_ALLOW_SHOP, false);
-                WorldGuard.getInstance().getFlagRegistry().register(allowShopFlag);
+                STATE_FLAG_OBJ_ALLOW_SHOP = new StateFlag(FLAG_ALLOW_SHOP, false);
+                registry.register(STATE_FLAG_OBJ_ALLOW_SHOP);
                 Bukkit.getLogger().log(Level.INFO,"[Shop] Registered WorldGuard flag '" + FLAG_ALLOW_SHOP + "'");
             } catch (FlagConflictException | IllegalStateException e) {
                 // Another plugin has probably already registered this flag,
                 // or this plugin got hard reloaded by some plugin manager plugin.
+                Flag<?> existing = registry.get(FLAG_ALLOW_SHOP);
+                if (existing instanceof StateFlag) {
+                    WorldGuardHook.STATE_FLAG_OBJ_ALLOW_SHOP = (StateFlag) existing;
+                    Bukkit.getLogger().log(Level.INFO,"[Shop] WorldGuard flag already registered '" + FLAG_ALLOW_SHOP + "'");
```

Comment:
We should be using the internal logger "this.getLogger()" instead of Bukkit.getLogger()